kernel/src/net/socket/ip/stream/connected.rs

```diff
@@ -45,12 +55,23 @@ impl ConnectedStream {
             bound_socket,
             remote_endpoint,
             is_new_connection,
+            is_receiving_closed: AtomicBool::new(false),
+            is_sending_closed: AtomicBool::new(false),
         }
     }
 
-    pub fn shutdown(&self, _cmd: SockShutdownCmd) -> Result<()> {
-        // TODO: deal with cmd
-        self.bound_socket.close();
+    pub fn shutdown(&self, cmd: SockShutdownCmd, pollee: &Pollee) -> Result<()> {
+        if cmd.shut_read() {
+            self.is_receiving_closed.store(true, Ordering::Relaxed);
```

Comment:
I think we need to make some adjustments to `try_read`. If `is_receiving_closed` is true, we can't return `Errno::EAGAIN` in `try_read`, otherwise we will do an infinite loop if the socket is in blocking mode.

However, no changes are needed for `try_write`. This is because we call `bound_socket.close()` after setting `is_sending_closed`.

Reply:
Yes, if `is_receiving_closed` is true, the socket should return `Ok(0)` instead of `Err(EAGAIN)`. The code has been refactored and the test case has been added.